lib/rules/time-min-milliseconds/index.js

```diff
@@ -62,47 +62,44 @@ const rule = (primary, secondaryOptions) => {
 			}
 
 			if (shorthandTimeProperties.has(propertyName)) {
-				const valueListList = postcss.list.comma(propertyValue);
-
-				for (const valueListString of valueListList) {
-					const valueList = postcss.list.space(valueListString);
-
-					if (ignoreDelay) {
-						// Check only duration time values
-						const duration = getDuration(valueList);
-
-						if (duration && !isAcceptableTime(duration)) {
-							complain(decl, propertyValue.indexOf(duration), duration.length);
-						}
-					} else {
-						// Check all time values
-						for (const value of valueList) {
-							if (!isAcceptableTime(value)) {
-								complain(decl, propertyValue.indexOf(value), value.length);
-							}
-						}
-					}
-				}
+				const parsedValue = valueParser(getDeclarationValue(decl));
+				let isValueTimeCount = 0;
+
+				parsedValue.walk(({ value, sourceIndex }) => {
+					isValueTimeCount = valueTimeCheck(value, isValueTimeCount);
+
+					if (isAcceptableTime(value) || (ignoreDelay && isValueTimeCount !== 1)) return;
```

Comment:
[suggest] I'm a bit concerned about performance degradation due to extra calls of `valueParse.unit()` inside `valueTimeCheck()` and `isAcceptableTime()`. How about passing a `import('postcss-value-parser').Dimension` object instead of a value string like this?

```js
const dimension = valueParser.unit(value);
isValueTimeCount = valueTimeCheck(dimension, value, isValueTimeCount);

if (isAcceptableTime(dimension) || (ignoreDelay && isValueTimeCount !== 1)) return;
```

Reply:
Agreed. I suggest we also use the [`getDimension()`](https://github.com/stylelint/stylelint/blob/main/lib/utils/getDimension.js) util rather than use `valueParser.unit()` directly as it's safer for non-standard syntax.

Reply:
@ybiquitous @jeddy3 
Thanks for the great advice!
I think these fixed can be done by tomorrow.